examples/utils/initialize_settings.py

```diff
@@ -37,26 +37,54 @@ def set_notebook_environment(environment_variables_config):
     """
     notebook_environment = environment_variables_config["notebook_environment"]
     if notebook_environment == "Colab":
-        from utils.colab import setup_colab_runtime_environment
+        from examples.utils.colab import setup_colab_runtime_environment
 
         setup_colab_runtime_environment(environment_variables_config)
     else:
-        from utils.generic import ensure_packages_are_installed
+        from examples.utils.generic import ensure_packages_are_installed
 
         ensure_packages_are_installed(notebook_environment)
 
 
-def get_notebook_name():
+def get_notebook_info():
     """
     Get the name of a currently running notebook in Google Colab.
     Args:
         None
     Return:
-        filename
+        a dict with notebook info.
     """
-    ip = socket.gethostbyname(socket.gethostname())  # 172.28.0.12
-    filename = requests.get(f"http://{ip}:9000/api/sessions").json()[0]["name"]
-    return filename
+    # ip = socket.gethostbyname(socket.gethostname())  # 172.28.0.12
+    ip = os.getenv("COLAB_JUPYTER_IP")
+    response = requests.get(f"http://{ip}:9000/api/sessions").json()[0]
+
+    notebook_name = response["name"]
+    path = urlparse(unquote(response["path"]).split("=")[1]).path
+    parsed = re.findall("(.*)/blob/(.*)/examples/(.*)", path)[0]
+    github_org_repo = parsed[0][1:]  # remove leading /
+    branch_name = parsed[1]
+    notebook_path = f"examples/{parsed[2]}"
+
+    return dict(
+        notebook_name=notebook_name,
+        notebook_path=notebook_path,
+        branch_name=branch_name,
+        github_org_repo=github_org_repo,
+    )
+
+
+def get_notebook_path():
+    """
+    A proxy function used for a single string return when
+    the corresponding entry-point script in 'setup.py' is called.
+
+    Args:
+        None
+    Return:
+        the path of the notebook being currently executed.
```

Comment:
I find this a bit misleading, the function actually returns `None` but prints the path

Reply:
You are right, I went back and forth with the function between the `return` and `print`, and forgot to update the docstring. My comment below indicates that the `return` approach results in the non-zero return code when the function is executed via the entry point in `setup.py`, so I switched to the `print` approach. I'll fix the docstring.

Reply:
Addressed via 6078b14.